state/redis/redis.go

```diff
@@ -241,6 +311,53 @@ func (r *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
 	}, nil
 }
 
+func (r *StateStore) getJSON(req *state.GetRequest) (*state.GetResponse, error) {
+	res, err := r.client.Do(r.ctx, "JSON.GET", req.Key).Result()
+	if err != nil {
+		return nil, err
+	}
+
+	str, ok := res.(string)
```

Comment:
similar to line 295 should there be a nil check for `res`? 
I see that it is being typecasted to string, and that should return an empty string for nil values, but the return value from this function is different than what it is from the getDefault function ...